api/nodes/serializers.py

```diff
@@ -396,8 +397,14 @@ def create(self, validated_data):
             raise InvalidModelValueError(detail=e.messages[0])
         if len(tag_instances):
             node.tags.add(*tag_instances)
+
+        if is_truthy(request.GET.get('inherit_tags')) and validated_data['parent'].has_permission(user, 'write'):
+            parent = validated_data['parent']
+            for tag in parent.tags.filter():
+                node.tags.add(tag)
```

Comment:
You should just be able to use `node.add_tag(...)` here. It handles the addition and creates the log for you.